cmd/prometheus/main.go

```diff
@@ -244,8 +245,7 @@ func (c *flagConfig) setFeatureListOptions(logger *slog.Logger) error {
 				}
 				logger.Info("Enabled automatic configuration file reloading. Checking for configuration changes every", "interval", c.autoReloadInterval)
 			case "auto-gomemlimit":
-				c.enableAutoGOMEMLIMIT = true
-				logger.Info("Automatically set GOMEMLIMIT to match Linux container or system memory limit")
+				logger.Warn("Automatically set GOMEMLIMIT to match Linux container or system memory limit (deprecated, ignored)")
```

Comment:
3.0 would also be a chance to remove the feature flag completely.
Logging on unknown feature flags happens in the default branch.

Reply:
Sounds good, removed.